src/rules/continuous_transition/a.jl

```diff
@@ -0,0 +1,19 @@
+@rule ContinuousTransition(:a, Marginalisation) (q_y_x::MultivariateNormalDistributionsFamily, q_a::MultivariateNormalDistributionsFamily, q_W::Any, meta::CTMeta) = begin
+    ma = mean(q_a)
+    mW = mean(q_W)
+    myx, Vyx = mean_cov(q_y_x)
+
+    dy, dx = getdimensionality(meta)
+
+    mx, Vx = @views myx[(dy + 1):end], Vyx[(dy + 1):end, (dy + 1):end]
+    my, Vy = @views myx[1:dy], Vyx[1:dy, 1:dy]
+    Vyx    = @view Vyx[1:dy, (dy + 1):end]
+
+    Fs, es = getjacobians(meta, ma), getunits(meta)
+
+    # rank1update(Vyx, mx, my) equivalent to ξ = (Vyx + mx * my') 
+    D = sum(sum(es[j]' * mW * es[i] * Fs[i]' * rank1update(Vx, mx) * Fs[j] for i in 1:length(Fs)) for j in 1:length(Fs))
```

Comment:
I think it would be possible to refactor it to something like
```julia
    D = 0.0
    tmp = rank1update(Vx, mx)
    for i in 1:length(Fs)
        for j in 1:length(Fs)
            D += mW[j,i] * dot(Fs[i], tmp, Fs[j])
        end
    end
```
Might not look that great, but I think it will be a lot faster

Reply:
Also noted similar patterns in other places of the code. Try to make use of triple dot products where possible, and if it concerns standardbasisvectors, then just index the matrix

Reply:
Thanks @bartvanerp. I will try to optimize this part, however dot won't work here. Note, that D is a matrix, not a scalar.